File: src/main/java/no/novari/flyt/integration/validation/constraints/ActiveConfigurationIsDefinedIfStateIsActiveValidator.java
package no.novari.flyt.integration.validation.constraints;

import no.novari.flyt.integration.model.entities.Integration;
import no.novari.flyt.integration.validation.IntegrationValidationContext;
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;

public class ActiveConfigurationIsDefinedIfStateIsActiveValidator implements
        HibernateConstraintValidator<ActiveConfigurationIsDefinedIfStateIsActive, Integration.State> {

    @Override
    public boolean isValid(Integration.State value, HibernateConstraintValidatorContext hibernateConstraintValidatorContext) {
        return value != Integration.State.ACTIVE ||
                hibernateConstraintValidatorContext
                        .getConstraintValidatorPayload(IntegrationValidationContext.class)
                        .getConfiguration() != null;
    }

}

File: src/main/kotlin/no/novari/flyt/integration/validation/constraints/ActiveConfigurationIsDefinedIfStateIsActiveValidator.kt
package no.novari.flyt.integration.validation.constraints

import no.novari.flyt.integration.persistence.entity.Integration
import no.novari.flyt.integration.validation.IntegrationValidationContext
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext

class ActiveConfigurationIsDefinedIfStateIsActiveValidator :
    HibernateConstraintValidator<ActiveConfigurationIsDefinedIfStateIsActive, Integration.State> {
    override fun isValid(
        value: Integration.State?,
        hibernateConstraintValidatorContext: HibernateConstraintValidatorContext,
    ): Boolean {
        return value != Integration.State.ACTIVE ||
            hibernateConstraintValidatorContext
                .getConstraintValidatorPayload(IntegrationValidationContext::class.java)
                ?.configuration != null
    }
}
